nfs/src/rpc_task.cpp

```diff
@@ -1313,30 +1273,49 @@ void rpc_task::run_write()
      * membufs. We don't wait for the writes to actually finish, which means
      * we support buffered writes.
      */
-    std::vector<bytes_chunk> bc_vec =
-            copy_to_cache(this, ino, bufv, offset, length, error_code);
+    uint64_t periodic_bytes = 0;
+
+    copy_to_cache(this, ino, bufv, offset, length, error_code, &extent_left, &extent_right);
     if (error_code != 0) {
         AZLogWarn("[{}] copy_to_cache failed with error={}, "
                   "failing write!", ino, error_code);
         reply_error(error_code);
         return;
     }
 
-    assert(bc_vec.size() > 0);
+    assert(extent_right >= extent_left);
 
-    for (bytes_chunk& bc : bc_vec) {
-        /*
-         * Create the flush task to carry out the write.
-         */
-        struct rpc_task *flush_task =
-            get_client()->get_rpc_task_helper()->alloc_rpc_task(FUSE_FLUSH);
-        flush_task->init_flush(nullptr /* fuse_req */, ino);
+    /*
+     * Get the prune goals, if it's more than data already in-flight
+     * data to BLOB, then issue fresh flush.
+     */
+    inode->filecache_handle->get_prune_goals(nullptr, &periodic_bytes);
+    AZLogDebug("periodic_bytes: {}, extent_left{}, extent_right: {}",periodic_bytes, extent_left, extent_right);
+
+    if (inode->inflight_bytes >= periodic_bytes) {
+        AZLogDebug("Reply write without syncing to Blob");
+        reply_write(length);
+        return;
+    }
+
+    /*
+     * TODO: It may happen extent_left and extent_right smaller than
+     *       periodic_bytes - inode->inflight, then we need to go beyond
+     *       that range or try find continous chunk larger/multiple chunk
+     *       which meet our prune goals. It should not be required for sequential
+     *       but for random we can use this to make it sequential (That will be the
+     *       benefit for cache).
+     */
+    std::vector<bytes_chunk> bc_vec =
+        inode->filecache_handle->get_dirty_bc_range(extent_left, extent_right);
```

Comment:
actually now extent_left and extent_right doesn't make sense. Since we are flushing not when the current extent >= wsize but when we have enough dirty data in the inode. So asking for dirty bcs only in this range isn't right.
I think you try to say the same using the above TODO?